Formula/v/veryfasttree.rb

```diff
@@ -0,0 +1,137 @@
+class Veryfasttree < Formula
+  desc "Efficient phylogenetic tree inference for massive taxonomic datasets"
+  homepage "https://github.com/citiususc/veryfasttree"
+  url "https://github.com/citiususc/veryfasttree/archive/refs/tags/v4.0.3.tar.gz"
+  sha256 "770e41773806c65e60ebc77363beccf40e3d339affd97c07eb7b2e1acb51af60"
+  license "GPL-3.0-only"
+  head "https://github.com/citiususc/veryfasttree.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cmake" => :build
+  depends_on "gcc"
+
+  fails_with :clang do
+    cause "Apple clang is not supported"
+  end
+
+  fails_with :gcc do
+    version "4" # fails with GCC 4.x and earlier
+    cause "Requires C++11 full support gcc >=5"
+  end
+
+  def install
+    script = <<~EOS
+      #!/bin/sh
+
+      BASE=LIBEXEC_PATH/VeryFastTree
+
+      test_and_run () {
+          if ARCH_CHECK && [ -x "${BASE}-$1" ]; then
+              cmd="${BASE}-$1"
+              shift
+              exec "${cmd}" "$@"
+          fi
+      }
+
+      for SIMD in avx512f avx2 avx sse4 ; do test_and_run ${SIMD} "$@" ; done
+
+      # fallback to base option
+      exec "${BASE}-sse2" "$@"
+    EOS
```

Comment:
????

Reply:
This snippet of the formula, along with the one you mentioned later, is part of a strategy for packaging binaries with different optimizations. This solution was suggested to me when packaging VeryFastTree for [Debian](https://lists.debian.org/debian-med/2023/07/msg00011.html). There, the baseline for compilation is SSE2, but when application performance depends on using more modern extensions, they recommend packaging different binaries and then selecting the most suitable one at runtime. This only occurs when the user downloads the binary; in case they compile it themselves, CMake would select the most appropriate version for their hardware.

I'm not sure if there's a better way to do it in Homebrew, but compiling with minimal options will result in a lack of performance on more modern computers.

Reply:
Homebrew, in stark contrast with Debian, goes to great lengths to keep the build as close to the upstream configuration as possible (see also: https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-heavy-manual-prepost-install-intervention). So this is not an acceptable way to package software for Homebrew.

> compiling with minimal options will result in a lack of performance on more modern computers

That is up to upstream to resolve.